inout.go

```diff
@@ -163,22 +163,27 @@ import "go.uber.org/dig"
 // Note that values in a value group are unordered. Fx makes no guarantees
 // about the order in which these values will be produced.
 //
+// # Soft Value Groups
+//
+// A soft value group can be thought of as a best-attempt at populating the
+// group with values from constructors that have already run. In other words,
+// if a constructor's output type is only consumed by a soft value group,
+// it will not be run.
```

Comment:
Do you think that it should clarify that fx doesn't guarantee to populate the soft value groups with all the values from constructors that are called after the creation of te `fx.App`, that it only guarantees to not call constructors that only provides values to the soft group?

Reply:
👍 Sg. Let me update that.